Render exception when there is another Surface that also calls eglMakeCurrent

If an activity includes another Surface that also calls `eglMakeCurrent()` the app shows a number of `EGL_BAD_DISPLAY` crashes with the following exception on the Java side:

```
Caused by: java.lang.Error: eglSwapBuffers() failed
at com.mapbox.mapboxsdk.maps.NativeMapView.nativeRender(Native Method)
at com.mapbox.mapboxsdk.maps.NativeMapView.render(SourceFile:177)
```

This seems to be due to not resetting the `firstRender` flag in [`native_map_view.cpp`](https://github.com/mapbox/mapbox-gl-native/blob/a9e09c710e6b07be425640bd323200963e22d825/platform/android/src/native_map_view.cpp#L1401). Resetting it to `true` on `destroySurface()` fixes it.

Additionally, we don't have any test activities with another (GL) surface to test this situation. We should add a new one to the Test App.

Possibly related: https://github.com/mapbox/mapbox-gl-native/issues/6647.

cc: @ivovandongen @tobrun

--- a/platform/android/src/native_map_view.cpp
+++ b/platform/android/src/native_map_view.cpp
@@ -1416,6 +1416,7 @@ void NativeMapView::_destroySurface() {
     }
 
     surface = EGL_NO_SURFACE;
+    firstRender = true;
 
     if (window != nullptr) {
         ANativeWindow_release(window);

